### PR TITLE
server: Improved code coverage for policies and user model

### PR DESCRIPTION
**Pull request by aufdetim**
added tests to improve code coverage on app/policies/application_policy.rb, app/models/user.rb, app/policies/library_entry_policy.rb, app/policies/user_policy.rb.

### CODE REVIEW

File: server/spec/policies/user_policy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  subject { described_class }
+
+  permissions :show? do
+    it ('should allow anons') { should permit(nil, other) }
+    it ('should allow users') { should permit(user, other) }
+  end
+
+  permissions :create? do
+    it ('should allow admins') { should permit(admin, other) }
+    it ('should allow normal users') { should permit(user, other) }
```
Comment (NuckChorris) [inline server/spec/policies/user_policy_spec.rb]: This should probably actually be denied, now that I look at it in spec form. Users shouldn't be able to make other users, that's just weird

**Comment by NuckChorris** [comment]
You, sir, are awesome. Specs look fantastic, except that one behavior on user creation (which probably would've been unnoticed without this!)

**Comment by aufdetim** [comment]
Is there anything you want me to change? I can switch the spec to expect users to not have that permission, but then the test would fail. I'm not very familiar with how the policies work so I wouldn't be confident fixing it.

**Comment by NuckChorris** [comment]
It's not difficult, and the specs are there to build confidence! It's probably as easy as making `create?` return `!user`

**Comment by NuckChorris** [comment]
Huge props for the coverage boost 🎉

**Pull request merged**